src/status_im2/contexts/onboarding/common/carousel/animation.cljs

```diff
@@ -34,21 +31,21 @@
     paused)))
 
 (defn initialize-animation
-  []
+  [progress paused]
```

Comment:
refactored carousel so we can use it on any page now 👍

Reply:
I don't see much benefit in doing this. This complicates stuff instead of simplifying and not sure we even have several slide animations like this in the app. I would recommend just-copy pasting the carousel code and modifying it for your use case.

Reply:
> It's a common discussion I've seen in many other PRs, perhaps a guideline should added

Maybe we create a file with name general recommendations and suggestions and move some of the stuff there (the guideline sounds a little strong)

Having local atoms is either always not feasible or sometimes makes things complicated and/or creates problems. As in this case, the problems it creates or might create are mentioned in detail in the below [comment](https://github.com/status-im/status-mobile/pull/15684#discussion_r1171036317) and original [implementation](https://github.com/status-im/status-mobile/pull/15489/files#r1150463436).

I agree with you, components should have a local state. Still, we have several global atoms because they are simpler(and sometimes better) solutions.

Reply:
Probably @J-Son89 will find a solution for using a local atom in this case and solve the all above-mentioned problem, But is it worth the time he will spend in the process?

Reply:
Good points @Parveshdhull, some recommendations might be too strong sometimes.

Reply:
So I got this working, I made  use of global atoms however I put these in the background file as these are the backgrounds carousel atoms. Everything else remains pure functions then so the underlying functionality is completely reusable. 
Basically if a page now wants to use a carousel it has to handle the setup, i.e ratoms etc and then initialise and render the carousel.